pkg/monitor/etcd.go

```diff
@@ -0,0 +1,73 @@
+package monitor
```

Comment:
Can this entire logic be better rolled into the product by having etcd operator write an event on leader change?